### ISSUE TITLE
[Mobile Support] The page never links to install mobile app, and is missing an "open app" button

### ISSUE THREAD

**Issue by BillCarsonFr** [matrix-org/matrix.to#165]
Currently on a mobile browser if you click on a matrix.to link and you don't have the app, it will bring you to the web client instead of offer you to install the app.

* Actual:
<img width="282" alt="image" src="https://user-images.githubusercontent.com/9841565/98347596-e39d3700-2017-11eb-99fe-78fca5ea6599.png">

And accept opens the web client on mobile

* Expected (if android detected)
<img width="175" alt="image" src="https://user-images.githubusercontent.com/9841565/98347623-ed269f00-2017-11eb-9b8a-a9fa707f9b45.png">

** Second Issue on Mobile

=> Mobile app need to deep link matrix.to into the app, It is currently possible on android, but for iOS the matrix.to page should host an `apple-app-site-association` file. 
The issue with that is that we should host that file with info for all existing clients (the site-association contains the app id)

I think it would probably better to have a openApp button using a custom scheme `matrixto://@...` ( like mailto://)

=> This will allow any mobile client to handle it, and let the OS ask users what app to use if there are several

**Comment by BillCarsonFr** [comment]
TLDR;
I'd expect to buttons on mobile: 
- A button `Install App` [playstore + Fdroid] (android or iphone based on UA) + add userId/RoomAliaas.. as additional `referrer` query param
- A button `Open In App` with url matrixto://

(Unfortunatly there is no reliable way to know if the app is installed, so we have to show both)

**Comment by ara4n** [comment]
>  for iOS the matrix.to page should host an apple-app-site-association file.

I'm pretty sure it used to. I suspect it got lost during the upgrade 👍 

> I think it would probably better to have a openApp button using a custom scheme matrixto://@... ( like mailto://)

We have an official IANA matrix:// URI scheme proposal, as per https://github.com/quotient-im/matrix-doc/blob/kitsune/matrix-uri/proposals/2312-matrix-uri.md - but it's still in review.  Until it's fixed, I'd be inclined to just use the `apple-app-site-association` approach.

**Comment by bwindels** [comment]
We can probably generate `apple-app-site-association ` as a build artefact by going over the client map and see which ones have iOS in their platforms list. We could add a platformId callback to `LinkedClient` with the platform passed in to determine the platform specific client id. Happy to help with this.

**Comment by t3chguy** [comment]
Doesn't solve the issue in the case of element ios forks which would also want to handle matrix links, the matrix scheme is a more apt solution

**Comment by bwindels** [comment]
Yep, `apple-app-site-association` is an interim solution until the matrix scheme is a thing.

**Comment by bwindels** [comment]
Notes after doing handover with Valere:

 - multiple store install button with correct look for platform (apple store, play store, fdroid)
 	- mostly working on valr branch, just looks for button use npm module that gives typescript
 - deferred deeplinking:
 	- android: query parameter play store link
 	- ios: shared cookie

 - 1 click to install (on top) 
 - remember checkbox doesn't make sense 
 - for mobile: disable accept invite or rename to open in app
 	- for both ios and android, the open in app button should just refresh the page
 - matrix.to links will be more prominent in the next android release
 - valr/mobile_support
 - install buttons most important

**Comment by BillCarsonFr** [comment]
Closing as install buttons are there
Deferred deep link, open in app are moved to a new task
